frontend/src/pages/profile/components/avatar-wrapper/avatar-wrapper.tsx

```diff
@@ -0,0 +1,110 @@
+import {
+  useAppDispatch,
+  useCallback,
+  useHover,
+  useRef,
+  useState,
+} from '~/libs/hooks/hooks.js';
+import { type UserAuthResponseDto } from '~/packages/users/users.js';
+import { SUPPORTED_FILE_TYPES_STRING } from '~/pages/profile/constants/constants.js';
+import { actions as filesActions } from '~/slices/file/file.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  user: UserAuthResponseDto;
+  errorImageUpload: string | null;
+  className?: string;
+  onRemoveAvatar: () => void;
+  onUpdateAvatarId: (value: number | null) => void;
+  onErrorUpload: (value: string | null) => void;
+};
+
+const AvatarWrapper: FC<Properties> = ({
+  user,
+  errorImageUpload,
+  onRemoveAvatar,
+  onUpdateAvatarId,
+  onErrorUpload,
+}): JSX.Element => {
+  const dispatch = useAppDispatch();
+  const [previewUrl, setPreviewUrl] = useState<string | null>(user.avatarUrl);
+  const imageGroupInnerReference = useRef(null);
+  const isHover = useHover(imageGroupInnerReference);
+
+  const handleUploadUserAvatar = useCallback(
+    (event_: ChangeEvent<HTMLInputElement>): void => {
+      const [image] = event_.target.files ?? [];
+
+      if (image) {
+        const formData = new FormData();
+        formData.append('file', image);
+
+        void dispatch(filesActions.uploadFile(formData))
+          .unwrap()
+          .then((fileData) => {
+            onUpdateAvatarId(fileData.id);
+            setPreviewUrl(fileData.url);
+            onErrorUpload(null);
+          })
+          .catch((error: Error) => {
+            onErrorUpload(error.message);
+          });
+      }
+    },
+    [dispatch, onUpdateAvatarId, onErrorUpload],
+  );
+
+  const handleRemoveAvatar = useCallback(() => {
+    onRemoveAvatar();
+    setPreviewUrl(null);
+    onErrorUpload(null);
+  }, [onErrorUpload, onRemoveAvatar]);
+
+  return (
+    <div className={styles.imageGroup}>
+      <div className={styles.imageGroupInner} ref={imageGroupInnerReference}>
+        <label className={styles.imageWrapper} htmlFor="avatarId">
+          <Avatar
+            username={`${user.firstName} ${user.lastName}`}
+            avatarUrl={previewUrl}
+            className={styles.avatar}
+          />
+          <input
+            className="visually-hidden"
+            id="avatarId"
+            onChange={handleUploadUserAvatar}
+            type="file"
+            accept={SUPPORTED_FILE_TYPES_STRING}
+          />
+          <span
+            className={getValidClassNames(
+              styles.iconButtonEdit,
+              styles.iconButton,
+            )}
+          >
+            <Icon iconName="edit" className={styles.iconEdit} />
+          </span>
+        </label>
+        <IconButton
+          iconName="crossMark"
+          className={getValidClassNames(
+            styles.iconButtonRemove,
+            styles.iconButton,
+            previewUrl && isHover && styles.iconButtonRemoveActive,
```

Comment:
Just wondering why can’t we control hover styles using css `:hover` pseudo class?

Reply:
Yep, I believe we don't need separate hook to track hovering

Reply:
Oh, thanks. that's a leftover from my attempt to implement this without an additional wrapper. I probably saw what worked and didn't think to make it simple. Thanks for noticing.

- [5d09f35](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/pull/118/commits/5d09f3543d937477cab2ce9aca7f40ab32b51eb5)
- [a2d7299](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/pull/118/commits/a2d72992745623127dafbecda5792f23af2acb4f)